crates/matrix-sdk/src/widget/mod.rs

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! Widget API implementation.
+#![doc = include_str!("README.md")]
```

Comment:
Any reason you prefer this to be in a separate file? We do this sometimes when the docs get really big, but this seems to be manageable?

Reply:
It might get a bit bigger and Since I was also planning to add the ascii graphic this seemed appropriate.
It is now closer to what I expect it to become. If you still feel like it is short enough to be inlined I can totally move it here.

Reply:
Yeah, 200 lines is still reasonable enough.

Reply:
The other aspect is that its much easier to work with the docstring if its a seperate file and we get support by the editors markdown tooling.

Reply:
Oh, that. But my editor supports markdown tooling even if the markdown is inside of a Rust file.

I guess if it makes your life much easier then I'm fine with leaving it in a separate file.

Reply:
Cool, that we can keep it in a seperate file. This made the diagram work mach easier. I copied updated version in a lot and always fixing the docstring `///` would have been an annoying overhead.